packages/client3/components/nav/wallet-button.tsx

```diff
@@ -1,43 +1,119 @@
-import { Button, Popover } from "@/components/design-system";
+import detectEthereumProvider from "@metamask/detect-provider";
+import React, { useCallback, useEffect, useState } from "react";
+
+import { ButtonType, IconProps, Tooltip } from "@/components/design-system";
 import { DESIRED_CHAIN_ID } from "@/constants";
-import { openWalletModal } from "@/lib/state/actions";
+import { useCurrentUserWalletInfoQuery } from "@/lib/graphql/generated";
 import { useWallet } from "@/lib/wallet";
+import { metaMask } from "@/lib/wallet/connectors/metamask";
 
-import { Identicon } from "../identicon";
-import { WalletStatus } from "./wallet-status";
+import { Button } from "../design-system";
+import { User, ButtonStateText, IWalletButtonStyles } from "./types";
 
-export function WalletButton() {
-  const { account, error, connector } = useWallet();
+export function WalletButton({
+  getAccount: getAccount,
+}: {
+  getAccount: (account: User) => void;
+}) {
+  const { account, isActive, error } = useWallet();
+  const [isLoading, setIsLoading] = useState(false);
 
-  return error ? (
-    <Button
-      className="text-clay-500 h-10"
-      iconRight="Exclamation"
-      onClick={
-        error.name === "ChainIdNotAllowedError"
-          ? () => connector.activate(DESIRED_CHAIN_ID)
-          : openWalletModal
+  const detectBrowser = useCallback(async () => {
+    const provider = await detectEthereumProvider();
+    if (provider) {
+      return true;
+    }
+    return false;
+  }, []);
+
+  const handleWalletConnect = async () => {
+    setIsLoading(true);
+    if (await detectBrowser) {
+      if (!isActive) {
+        await metaMask.activate(DESIRED_CHAIN_ID);
       }
-    >
-      {error.name === "ChainIdNotAllowedError"
-        ? "Wrong network"
-        : "Wallet error"}
-    </Button>
-  ) : account ? (
-    <Popover
-      placement="bottom-end"
-      content={({ close }) => <WalletStatus onWalletDisconnect={close} />}
-    >
-      <Button className="inline-flex h-10 items-center gap-3 !px-2 md:!px-4">
-        <span className="hidden md:block">
-          {account.substring(0, 6)}...{account.substring(account.length - 4)}
-        </span>
-        <Identicon account={account} scale={3} />
-      </Button>
-    </Popover>
+    }
+    setIsLoading(false);
+  };
+
+  function handleAddressFormat(account: string) {
```

Comment:
Would recommend extracting this function to a utils class to provide more reusability and also would recommend using arrow function